pkg/cortex/cortex.go

```diff
@@ -0,0 +1,111 @@
+// Package cortex provides methods to query data from our central
+// Prometheus data store.
+package cortex
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+)
+
+const (
+	query = `aggregation:giantswarm:cluster_release_version{pipeline="testing"}`
+)
+
+type Config struct {
+	// Base URL of the prometheus API server.
+	URL string
+	// User name.
+	UserName string
+	// Password (API token).
+	Password string
+}
+
+type Service struct {
+	config Config
+	client api.Client
+}
+
+func New(conf Config) (*Service, error) {
+	apiConfig := api.Config{
+		Address: conf.URL,
+		RoundTripper: config.NewBasicAuthRoundTripper(
+			conf.UserName,
+			config.Secret(conf.Password),
+			"",
+			api.DefaultRoundTripper),
+	}
+
+	client, err := api.NewClient(apiConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	s := &Service{
+		config: conf,
+		client: client,
+	}
+
+	return s, nil
+}
+
+// QueryClusters queries cortex for a list of workload clusters
+// that existed in time series at a given point in time.
+// Returns a sorted slice of strings with format "<installation>/<cluster_id>".
+func (s Service) QueryClusters(t time.Time) ([]string, error) {
+	clusters := []string{}
+
+	v1api := v1.NewAPI(s.client)
+
+	value, warnings, err := v1api.Query(context.Background(), query, t)
+	if err != nil {
+		return clusters, microerror.Mask(err)
+	}
+	if len(warnings) > 0 {
+		fmt.Printf("Warnings while querying: %#v\n", warnings)
+	}
+
+	if value == nil {
+		return clusters, microerror.Mask(fmt.Errorf("query %s returned nil value for time %s", query, t))
```

Comment:
```suggestion
		return clusters, microerror.Maskf(executionFailedError, "query %s returned nil value for time %s", query, t)
```

Could we use our regular error handling here? Just needs an executionFailedError to be definted.